qlib/pagetable.rs

```diff
@@ -664,6 +690,93 @@ impl PageTables {
         return Ok(res);
     }
 
+    #[cfg(feature = "cc")]
+    pub fn MapWith1G(
+        &self,
+        start: Addr,
+        end: Addr,
+        physical: Addr,
+        flags: PageTableFlags,
+        pagePool: &Allocator,
+        _kernel: bool,
+    ) -> Result<bool> {
+        if start.0 & (MemoryDef::HUGE_PAGE_SIZE_1G - 1) != 0
+            || end.0 & (MemoryDef::HUGE_PAGE_SIZE_1G - 1) != 0
+        {
+            panic!("start/end address not 1G aligned")
+        }
+
+        #[cfg(target_arch = "aarch64")]
+        let hugepage_flags = flags & (!PageTableFlags::TABLE);
+
+        #[cfg(target_arch = "x86_64")]
+        let hugepage_flags = flags | PageTableFlags::HUGE_PAGE;
+
+        let mut res = false;
+
+        let mut curAddr = start;
+
+        let identical = IDENTICAL_MAPPING.load(Ordering::Acquire);
+        let pt: *mut PageTable = if identical{
```

Comment:
Identical means identical mapping here, and it is a "cc" feature because unidentical mapping is enabled in CCMode NormalEmu, which is used to test whether cc modification works or not.

Reply:
do you mean the unidentical mapping does not work for non-cc (yet)?

If non-identical mapping is also planed for non-cc, then this feature doesn't need to be cc only. If the identical-mapping config breaks non-cc when turned on, then let it be.

Reply:
I think currently this is used only for cc.